src/main/java/dev/railroadide/railroad/ide/ui/MarkdownPreviewPane.java

```diff
@@ -0,0 +1,204 @@
+import javafx.geometry.Pos;
+import javafx.scene.Node;
+import javafx.scene.control.Button;
+import javafx.scene.control.ContextMenu;
+import javafx.scene.control.CustomMenuItem;
+import javafx.scene.control.SplitPane;
+import javafx.scene.layout.HBox;
+import javafx.scene.layout.Priority;
+import javafx.scene.layout.VBox;
+import javafx.scene.web.WebView;
+import org.kordamp.ikonli.Ikon;
+import org.kordamp.ikonli.fontawesome6.FontAwesomeBrands;
+import org.kordamp.ikonli.fontawesome6.FontAwesomeSolid;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+
+public class MarkdownPreviewPane extends RRVBox {
+    private final MarkdownWebView preview;
+    private final Path markdownFile;
+
+    private TextEditorPane textEditorPane;
+    private WebView webViewPane;
+
+    private final HBox topRow;
+    private final HBox markdownButtons;
+    private final HBox switchButtons;
+
+    public MarkdownPreviewPane(Path markdownFile) {
+        this.markdownFile = markdownFile;
+
+        try {
+            preview = new MarkdownWebView(Files.readString(markdownFile), Railroad.getHostServicess());
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+
+        markdownButtons = createMarkdownButtons();
+        HBox.setHgrow(markdownButtons, Priority.ALWAYS);
+        switchButtons = createViewButtons();
+
+        topRow = new HBox(markdownButtons, switchButtons);
+
+        showContent(splitView(), topRow);
+    }
+
+    private TextEditorPane codeView() {
+        if(textEditorPane != null)
+            return textEditorPane;
+
+        textEditorPane = new TextEditorPane(markdownFile);
+        textEditorPane.textProperty().addListener(
+            (observable, oldValue, newValue) -> preview.setContent(newValue));
+
+        return textEditorPane;
+    }
+
+    private SplitPane splitView() {
+        SplitPane splitPane = new SplitPane(codeView(), previewView());
+        // ensure divider is applied after layout to avoid a 0 width child
+        Platform.runLater(() -> splitPane.setDividerPosition(0, 0.5));
+        return splitPane;
+    }
+
+    private WebView previewView() {
+        if(webViewPane != null)
+            return webViewPane;
+
+        webViewPane = preview.launch();
+        return webViewPane;
+    }
+
+    // helper that replaces the content below the top row and makes it grow
+    private void showContent(Node content, HBox topRow) {
+        getChildren().clear();
+        getChildren().addAll(topRow, content);
+        VBox.setVgrow(content, Priority.ALWAYS);
+        textEditorPane.scrollToPixel(0, 0);
+    }
+
+    private HBox createViewButtons(){
+        Button codeView = createButton(FontAwesomeSolid.CODE);
+        Button splitView = createButton(FontAwesomeSolid.COLUMNS);
+        Button previewView = createButton(FontAwesomeBrands.MARKDOWN);
+
+        HBox switchButtons = new HBox(codeView, splitView, previewView);
```

Comment:
`RRHBox` should be used over `HBox` where possible.
```suggestion
        var switchButtons = new RRHBox(codeView, splitView, previewView);
```